docs/releases/slashing-release/interfaces/allocator.md

```diff
@@ -1,61 +1,277 @@
 # Allocator
 
-The `OperatorDetails` is modified to replace the `earningsReciever` field with the new `allocator` role that allocates slashable stake to operatorSets.
+TODO: Handle elimination of operator delegation and deposits
 
 ```solidity
 interface IDelegationManager {
 	/// STRUCTS
-	
-	// @notice Struct used for storing information about a single operator who has registered with EigenLayer
-	struct OperatorDetails {
-		// @notice address that can allocate slashable stake and register/deregister the operator to AVS operatorSets
-		address allocator;
-		/**
-		 * @notice Address to verify signatures when a staker wishes to delegate to the operator, as well as controlling "forced undelegations".
-		 * @dev Signature verification follows these rules:
-		 * 1) If this address is left as address(0), then any staker will be free to delegate to the operator, i.e. no signature verification will be performed.
-		 * 2) If this address is an EOA (i.e. it has no code), then we follow standard ECDSA signature verification for delegations to the operator.
-		 * 3) If this address is a contract (i.e. it has code) then we forward a call to the contract and verify that it returns the correct EIP-1271 "magic value".
-		 */
-		address delegationApprover;
-		// @notice noop
-		uint32 stakerOptOutWindowBlocks;
-	}
+
+	/**
+     * Struct type used to specify an existing queued withdrawal. Rather than storing the entire struct, only a hash is stored.
+     * In functions that operate on existing queued withdrawals -- e.g. completeQueuedWithdrawal`, the data is resubmitted and the hash of the submitted
+     * data is computed by `calculateWithdrawalRoot` and checked against the stored hash in order to confirm the integrity of the submitted data.
+     */
+    struct Withdrawal {
+        // The address that originated the Withdrawal
+        address staker;
+        // NEW: The staker's allocator at the time of thhe queing of the withdrawal
+        address allocator;
+        // The address that can complete the Withdrawal + will receive funds when completing the withdrawal
+        address withdrawer;
+        // Nonce used to guarantee that otherwise identical withdrawals have unique hashes
+        uint256 nonce;
+        // Block number when the Withdrawal was created
+        uint32 startBlock;
+        // Array of strategies that the Withdrawal contains
+        IStrategy[] strategies;
```

Comment:
Is it time to just reduce this to a single strategy? We already do this in `queueWithdrawal` where we have multiple withdrawal roots created and we know its going to be more expensive to complete withdrawals in the case of slashing events.

Reply:
it fucks up completion of existing withdrawals :(

unless we want to do a far more invasive upgrade where we upgrade to single strat withdrawals, wait a while, autocomplete all multistrategy withdrawals, and then remove autocomplete functions...

Reply:
Yup, basically the same thing we did with M2 withdrawals migration. I guess this will just remain as tech debt for the foreseeable future but it would be nice to address this at some point.